todo_app/app.py

```diff
@@ -1,11 +1,18 @@
-from flask import Flask
+from flask import Flask, redirect, render_template, request
+from todo_app.data.session_items import add_item, get_items
 
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
-@app.route('/')
+@app.route('/', methods=['GET', 'POST'])
 def index():
-    return 'Hello World!'
+    if request.method == 'GET':
+        items = get_items()
+        return render_template('index.html', items = items)
+    elif request.method == 'POST':
+        title = request.form.get('title')
+        add_item(title)
+        return redirect('/')
```

Comment:
Minor, but I'm a fan of splitting this into two methods - as these potentially get more complex they're handling quite different cases so it's arguably simpler to handle them one at a time, maybe like:

```python
@app.route('/', methods=['GET'])
def index():
    items = get_items()
    return render_template('index.html', items = items)

@app.route('/', methods=["POST"])
def create_new():
    title = request.form.get('title')
    add_item(title)
    return redirect('/')
```